python/cuml/manifold/umap.pyx

```diff
@@ -451,26 +460,57 @@ class UMAP(Base):
             y contains a label per row.
             Acceptable formats: cuDF Series, NumPy ndarray, Numba device
             ndarray, cuda array interface compliant array like CuPy
+        knn_indices : arra-like (device or host) shape=(n_samples, n_neighbors)
+            knn_indices contains the indices of nearest neighbors for
+            each sample. This parameter makes it possible to pre-compute, cache
+            and reuse a same KNN run. The user can also choose
```

Comment:
The ability to accept a KNN is not actually what's providing the feature for the user to select their own custom distance function- I want to make it clear to them that this gives them the option of using any distance function of their choosing, since the KNN provided in cuML currently only allows euclidean, and by using a different knn library, they can gain the benefit of using different distance metrics. We should be as clear as possible about this here in the docs so that the user understands the implications of doing this.

Reply:
Ok, I'll rewrite that in a clearer way.